sdk/core/Azure.Core/tests/ClientDiagnosticsTests.Net50.cs

```diff
@@ -468,6 +470,109 @@ public void FailedStopsActivityAndWritesExceptionEventActivitySource()
+            using TracerProvider OTelTracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource($"Azure.*")
+                .SetSampler(new TraceIdRatioBasedSampler(0.1))
+                .Build();
+
+            DiagnosticScopeFactory clientDiagnostics = new DiagnosticScopeFactory("Azure.Clients", "Microsoft.Azure.Core.Cool.Tests", true, true);
+
+            int activeActivityCounts = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                DiagnosticScope scope = clientDiagnostics.CreateScope("ClientName.ActivityName");
+                scope.Start();
+                if (Activity.Current.IsAllDataRequested) activeActivityCounts++;
+                scope.Dispose();
+            }
+
+            Assert.IsTrue(activeActivityCounts > 0, "No activities created");
+            Assert.IsTrue(activeActivityCounts < 20, "More than expected activities created");
+
+            Assert.IsNull(Activity.Current);
+        }
+
+        [Test]
+        [NonParallelizable]
+        public void OpenTelemetryCompatibilityWithAlwaysOffSampler()
+        {
+            using var _ = SetAppConfigSwitch();
+
+            // Open Telemetry Listener
+            using TracerProvider OTelTracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource($"Azure.*")
+                .SetSampler(new AlwaysOffSampler())
+                .Build();
+
+            DiagnosticScopeFactory clientDiagnostics = new DiagnosticScopeFactory("Azure.Clients", "Microsoft.Azure.Core.Cool.Tests", true, true);
+
+            int activeActivityCounts = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                DiagnosticScope scope = clientDiagnostics.CreateScope("ClientName.ActivityName");
+                scope.Start();
+                if (Activity.Current.IsAllDataRequested)
+                    activeActivityCounts++;
+                scope.Dispose();
+            }
+
+            Assert.AreEqual(0, activeActivityCounts);
+
+            Assert.IsNull(Activity.Current);
+        }
+
+        [Test]
+        [NonParallelizable]
+        public void OpenTelemetryCompatibilityWithCustomSampler()
+        {
+            using var _ = SetAppConfigSwitch();
+
+            // Open Telemetry Listener
+            using TracerProvider OTelTracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddSource($"Azure.*")
+                .SetSampler(new ParentBasedSampler(new CustomSampler()))
+                .Build();
+
+            DiagnosticScopeFactory clientDiagnostics = new DiagnosticScopeFactory("Azure.Clients", "Microsoft.Azure.Core.Cool.Tests", true, true);
+
+            int activeActivityCounts = 0;
+            for (int i = 0; i < 5; i++)
+            {
+                DiagnosticScope scope = clientDiagnostics.CreateScope("ClientName.ActivityName");
+                scope.AddLink($"00-6e76af18746bae4eadc3581338bbe8b{i}-2899ebfdbdce904b-00", "foo=bar");
+
+                scope.Start();
```

Comment:
```suggestion
                Assert.IsNull(scope.Start());
```

Reply:
Start doesn't return anything...why are we asserting is to null? or I am missing something here?